.github/workflows/ci_workflows.yml

```diff
@@ -110,3 +108,30 @@ jobs:
         - name: (Allowed Failure) Python 3.12 with remote data and dev version of key dependencies
           linux: py312-test-devdeps
           posargs: --remote-data=any --verbose
+
+  report_diff_coverage:
+    needs: [ tests ]
+    if: github.event_name == 'pull_request_target'
```

Comment:
using `pull_request_target` is necessary to have the correct permissions to comment on a PR; see https://github.com/spacetelescope/stpipe/actions/runs/9684229017/attempts/1#summary-26721558822 for a test of this same code in `spacetelescope/stpipe`

Reply:
Oh hmm... we cannot turn the CI workflow into pull_request_target for security reason. See https://securitylab.github.com/research/github-actions-preventing-pwn-requests/

Even if we split into 2 workflows, we have to be careful that the second one with elevated privilege will not be compromised by malicious coverage artifacts (is such a thing possible?).

Reply:
> we cannot turn the CI workflow into pull_request_target for security reason. See https://securitylab.github.com/research/github-actions-preventing-pwn-requests/

Ah ok, so if we remove the `pip install .` line then this will be safe to use `pull_request_target`, as then it would not be running *any* arbitrary code from the PR author.

Reply:
oh I see now, the *entire workflow* would have to be triggered on `pull_request_target`, which means that `pytest` would also be running there; oof

We'll have to figure out some other way of commenting on a PR; in the meantime I think we can just stick with the job summary table

Reply:
Alternatively, I can try and figure out how to make a newfangled Personal Access Token with fine-grained access only to comment on PRs; that would solve this issue

Reply:
Not sure what you mean by the fine-grained PAT. Would that need to be configured by every fork opening a PR? That would be too much to ask of contributors.

Yes, I think summary table is good enough for now, but still unclear to me at a glance if you are reporting project coverage, patch coverage, or both.

Reply:
> Not sure what you mean by the fine-grained PAT. Would that need to be configured by every fork opening a PR? That would be too much to ask of contributors.

you only need to create one, and add it to the secrets of the Astropy repository 

see here: https://github.com/settings/personal-access-tokens/new

you can set a PAT to only have access to select repositories:
[<img width="480" alt="image" src="https://github.com/astropy/astropy/assets/16024299/0ca1a0a2-0ade-4de3-afe3-c51c4fa60b51">](https://github.com/settings/personal-access-tokens/new)

and only permission to modify pull requests and nothing else:
[<img width="774" alt="image" src="https://github.com/astropy/astropy/assets/16024299/267cab9e-9577-4e6f-8c24-7d1d01176022">](https://github.com/settings/personal-access-tokens/new)


> Yes, I think summary table is good enough for now, but still unclear to me at a glance if you are reporting project coverage, patch coverage, or both.

Good point, I updated the job names to indicate overall or diff coverage